src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1488,6 +1488,90 @@ public async Task ValidateLocationHeaderFieldForPostRequests(EntitySourceType en
+        [DataRow(true, DisplayName = "Mutation operation fails when an extraneous field is included in request body and rest.request-body-strict is omitted from the rest runtime section in the config file.")]
+        public async Task ValidateStrictModeAsDefaultForRestRequestBody(bool includeExtraneousFieldInRequestBody)
+        {
+            string entityJson = @"
+            {
+                ""entities"": {
+                    ""Book"": {
+                        ""source"": {
+                        ""object"": ""books"",
+                        ""type"": ""table""
+                        },
+                        ""permissions"": [
+                        {
+                            ""role"": ""anonymous"",
+                            ""actions"": [
+                            {
+                                ""action"": ""*""
+                            }
+                            ]
+                        }
+                        ]
+                    }
+                }
+            }";
+
+            string configJson = TestHelper.AddPropertiesToJson(TestHelper.BASE_CONFIG, entityJson);
+            RuntimeConfigLoader.TryParseConfig(configJson, out RuntimeConfig deserializedConfig, logger: null, GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL));
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, deserializedConfig.ToJson());
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(SupportedHttpVerb.Post);
+                string requestBody = @"{
+                        ""title"": ""Harry Potter and the Order of Phoenix"",
+                        ""publisher_id"": 1234";
+
+                if (includeExtraneousFieldInRequestBody)
+                {
+                    requestBody += @",
+                    ""extraField"": 12";
+                }
+
+                requestBody += "}";
+                       
+
+                JsonElement requestBodyElement = JsonDocument.Parse(requestBody).RootElement.Clone();
+                HttpRequestMessage request = new(httpMethod, "api/Book")
+                {
+                    Content = JsonContent.Create(requestBodyElement)
+                };
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                if (includeExtraneousFieldInRequestBody)
+                {
+                    // Assert that including an extraneous field in request body while operating in strict mode leads to a bad request exception. 
+                    Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
```

Comment:
I think we don't need to validate the message here. The only difference between the two DataTests is the inclusion/exclusion of the extraneous field. One passes and the other fails with BadRequest error.

Reply:
> One passes and the other fails with BadRequest error.

Understood. Isn't the point to test that the request fails for a specific reason? If not testing the error code, could you test the substatus code in the response body?

Reply:
The reason I'm suggesting an extra validation is because 400 could have many root causes. We don't want this test to pass if the 400 is for something else (e.g. complaining about query string or URI component, or another request body component.)

Reply:
Put a check on the exception message.